src/duckling/canvas/drawing/util.ts

```diff
@@ -87,3 +103,22 @@ export function drawCanvasBorder(centerPosition : Vector, stageDimensions : Vect
         stageDimensions,
         graphics);
 }
+
+/**
+ * Converts an rgb color to a hex string
+ * @param  r red portion of the color (0-255)
+ * @param  g green portion of the color (0-255)
+ * @param  b blue portion of the color (0-255)
+ * @return hex string of the given rgb color (no # at the beginning)
+ */
+export function rgbToHex(r : number, g : number, b : number) : string {
```

Comment:
This could be a lot faster if you used arithmetic instead of string manipulation.
var red = (r & 255) << 16;
var green = (g & 255) << 8;
var blue = (b & 255);
return (red + green + blue).toString(16);